### DISCUSSION TITLE
Reboot/Snooze Buttons w/email notifications too...

### DISCUSSION THREAD

**Discussion by FLEXWKU** [Windos/BurntToast#222] [General]
I have created a toast that has a reboot button and a snooze button based upon selection of 5 mins, 10 mins....up to a day.  
the reboot calls a ps1 from registry that sends an email saying user clicked reboot, thus giving me notification of event.  and then the ps1 reboots the system.  However the snooze does not send an email while still keeping the selection options.  here is what i have thus far...please advise if you have any solutions.  thanx
is there an arguments (for the email) options for the snooze button while also keeping the selectionsbox options.

`
New-Item -Path "C:\" -Name "dploy" -ItemType "directory"
New-Item -ItemType "directory" -Path "C:\dploy\DoNotDelete\Toast_Reboot"
Copy-Item -Path "\\Server1\DoNotDelete\Toast_Reboot\SendRebootMail.ps1" -Destination "C:\dploy\DoNotDelete\Toast_Reboot"
Copy-Item -Path "\\Server1\DoNotDelete\Toast_Reboot\ToastRebootCmd.bat" -Destination "C:\dploy\DoNotDelete\Toast_Reboot"
Copy-Item -Path "\\Server1\DoNotDelete\Toast_Reboot\Logo.jpg" -Destination "C:\dploy\DoNotDelete\Toast_Reboot"
# *****************************************************************************************************

# Create Registry Location for EventLogs...
New-item -Path "HKLM:\SYSTEM\ControlSet001\Services\EventLog" -Name "BurntToast" -force
$newValue = New-ItemProperty -Path "HKLM:\SYSTEM\ControlSet001\Services\EventLog\BurntToast" -Name 'BurntToast' -PropertyType MultiString -Value @"
abcApplication
cdeApp
"@
$newValue.multistring

# *****************************************************************************************************

#Checking if ToastReboot:// protocol handler is present
# Deletes the key if it already existed, to make sure any changes made take effect...
reg delete  HKEY_CLASSES_ROOT\ToastReboot /f

New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT -erroraction silentlycontinue | out-null
$ProtocolHandler = get-item 'HKCR:\ToastReboot' -erroraction 'silentlycontinue'
if (!$ProtocolHandler) {
    #create handler for reboot
    New-item 'HKCR:\ToastReboot' -force
    set-itemproperty 'HKCR:\ToastReboot' -name '(DEFAULT)' -value 'url:ToastReboot' -force
    set-itemproperty 'HKCR:\ToastReboot' -name 'URL Protocol' -value '' -force
    new-itemproperty -path 'HKCR:\ToastReboot' -propertytype dword -name 'EditFlags' -value 2162688
    New-item 'HKCR:\ToastReboot\Shell\Open\command' -force
    # The CMD file below is used to execute a ps1 file in the same directory that sends an email and then starts a Reboot...
    set-itemproperty 'HKCR:\ToastReboot\Shell\Open\command' -name '(DEFAULT)' -value 'C:\dploy\DoNotDelete\Toast_Reboot\ToastRebootCmd.bat /q' -force
        
}

#******************************************************************************************************
Install-Module -Name BurntToast -Confirm:$false -Force
Install-module -Name RunAsUser -Confirm:$false -Force
invoke-ascurrentuser -scriptblock {
    
    $ComputerName = HostName
    $Domain = $Env:UserDomain
    $User = $Env:Username

    $heroimage = New-BTImage -Source 'C:\Dploy\DoNotDelete\Toast_Reboot\Logo.jpg' -HeroImage
    $Text1 = New-BTText -Content  "Message from IT"
    $Text2 = New-BTText -Content "IT has installed updates on your computer. Please select if you'd like to reboot now, or snooze this message."
       
    $Button = New-BTButton -Content "Snooze" -snooze -id 'SnoozeTime' 
    $Button2 = New-BTButton -Content "Reboot now" -Arguments "ToastReboot:" -ActivationType Protocol
    $5Min = New-BTSelectionBoxItem -Id 5 -Content '5 minutes' 
    $10Min = New-BTSelectionBoxItem -Id 10 -Content '10 minutes'
    $1Hour = New-BTSelectionBoxItem -Id 60 -Content '1 hour'
    $4Hour = New-BTSelectionBoxItem -Id 240 -Content '4 hours'
    $1Day = New-BTSelectionBoxItem -Id 1440 -Content '1 day'
    $Items = $5Min, $10Min, $1Hour, $4Hour, $1Day 
    $SelectionBox = New-BTInput -Id 'SnoozeTime' -DefaultSelectionBoxItemId 10 -Items $Items 
    $action = New-BTAction -Buttons $Button, $Button2 -inputs $SelectionBox     
    $Binding = New-BTBinding -Children $text1, $text2 -HeroImage $heroimage
    $Visual = New-BTVisual -BindingGeneric $Binding
    $Content = New-BTContent -Visual $Visual -Actions $action -Scenario IncomingCall
    Submit-BTNotification -Content $Content

}
`

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
